### PR TITLE
Generate new key function

### PR DESCRIPTION
**Pull request by weswhet**
Adding the ability to generate a new key on login with a single preference key 'GenerateNewKey', this key must not be set by a profile or MCX in order to work, and crypt will set it to False after successfully rotating the key.

### CODE REVIEW

**Comment by grahamgilbert** [comment]
Needs moar emoji.

**Review by grahamgilbert — APPROVED** [review]

**Comment by grahamgilbert** [comment]
Will merge when @mactroll has given his blessing.

**Comment by weswhet** [comment]
I made the requested change from the swift shaman, and he's given his blessing.

**Pull request merged**
